lib/rubocop/cop/style/redundant_line_continuations.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Avoid redundant line continuations.
+      #
+      # @example
+      #   # bad
+      #   foo. \
+      #     bar
+      #   foo \
+      #     .bar \
+      #       .baz
+      #   foo&. \
+      #     bar
+      #   [foo, \
+      #     bar]
+      #   {foo: \
+      #     bar}
+      #   foo(bar, \
+      #     baz)
+      #
+      #   # good
+      #   foo.
+      #     bar
+      #   foo
+      #     .bar
+      #       .baz
+      #   foo&.
+      #     bar
+      #   [foo,
+      #     bar]
+      #   {foo:
+      #     bar}
+      #   foo(bar,
+      #     baz)
+      #   foo('bar' \
+      #     'baz')
```

Comment:
These examples looks complicated. Can you update to simplify and clarify?

Reply:
Examples are organized.